File: TodoList/app/src/main/java/com/github/fredrik9000/todolist/add_edit_todo/DatePickerFragment.java
package com.github.fredrik9000.todolist.add_edit_todo;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.widget.DatePicker;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.Calendar;

public class DatePickerFragment extends DialogFragment
        implements DatePickerDialog.OnDateSetListener {

    private static final String TAG = "DatePickerFragment";
    private OnSelectDateDialogInteractionListener listener;

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // Use the current date as the default date in the picker
        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);

        // Create a new instance of DatePickerDialog and return it
        DatePickerDialog datePickerDialog = new DatePickerDialog(getActivity(), this, year, month, day);
        datePickerDialog.getDatePicker().setMinDate(System.currentTimeMillis() - 1000);
        return datePickerDialog;
    }

    @Override
    public void onDateSet(DatePicker view, int year, int month, int day) {
        listener.onSelectDateDialogInteraction(year, month, day);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        try {
            listener = (OnSelectDateDialogInteractionListener) getTargetFragment();
        } catch (ClassCastException e) {
            Log.e(TAG, "onAttach: ClassCastException : " + e.getMessage());
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }

    public interface OnSelectDateDialogInteractionListener {
        void onSelectDateDialogInteraction(int year, int month, int day);
    }
}

File: TodoList/app/src/main/java/com/github/fredrik9000/todolist/add_edit_todo/DatePickerFragment.kt
package com.github.fredrik9000.todolist.add_edit_todo

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.os.Bundle
import android.widget.DatePicker
import androidx.core.os.bundleOf
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.setFragmentResult
import java.util.*

class DatePickerFragment : DialogFragment(), OnDateSetListener {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        // Uses the current date as the default date in the picker
        with(Calendar.getInstance()) {
            return DatePickerDialog(requireActivity(), this@DatePickerFragment, this[Calendar.YEAR], this[Calendar.MONTH], this[Calendar.DAY_OF_MONTH]).apply {
                datePicker.minDate = System.currentTimeMillis() - 1000
            }
        }
    }

    override fun onDateSet(view: DatePicker?, year: Int, month: Int, day: Int) {
        setFragmentResult(DATE_PICKER_FRAGMENT_REQUEST_KEY, bundleOf(BUNDLE_YEAR_KEY to year, BUNDLE_MONTH_KEY to month, BUNDLE_DAY_KEY to day))
    }

    companion object {
        const val DATE_PICKER_FRAGMENT_REQUEST_KEY = "DATE_PICKER_FRAGMENT_REQUEST"
        const val BUNDLE_YEAR_KEY = "year"
        const val BUNDLE_MONTH_KEY = "month"
        const val BUNDLE_DAY_KEY = "day"
    }
}